jvector-base/src/main/java/io/github/jbellis/jvector/graph/disk/OnDiskGraphIndex.java

```diff
@@ -75,13 +78,16 @@ public class OnDiskGraphIndex implements ImmutableGraphIndex, AutoCloseable, Acc
     final NodeAtLevel entryNode;
     final int idUpperBound;
     final int inlineBlockSize; // total size of all inline elements contributed by features
-    final EnumMap<FeatureId, ? extends Feature> features;
+    final Map<FeatureId, ? extends Feature> features;
     final EnumMap<FeatureId, Integer> inlineOffsets;
+
     private final List<CommonHeader.LayerInfo> layerInfo;
     // offset of L0 adjacency data
     private final long neighborsOffset;
-    /** For layers > 0, store adjacency fully in memory. */
+    // For layers > 0, store adjacency fully in memory.
     private final AtomicReference<List<Int2ObjectHashMap<int[]>>> inMemoryNeighbors;
+    // When using fused features, store the features fully in memory for layers > 0
+    private final AtomicReference<Int2ObjectHashMap<FusedFeature.InlineSource>> inMemoryFeatures;
```

Comment:
It appears that these objects are not counted in the graph's `ramBytesUsed()` computations.

Reply:
Good catch!

Reply:
This comment is linked with https://github.com/datastax/jvector/pull/561#discussion_r2495627475. In order to give a correct estimation of the heap usage, we need to either load these structures at initialization or get the information from the header. I think it just males sense to to the former and solve this once and for all.